custom_components/tuya_local/devices/cleanair_604b_humidifier.yaml

```diff
@@ -0,0 +1,165 @@
+name: Humidifier
+products:
+  - id: cqoisddoci9i3yer
+    name: Clean Air Optima CA-604B Smart Top Filling
+primary_entity:
+  entity: humidifier
+  dps:
+    - id: 1
+      name: switch
+      type: boolean
+      mapping:
+        - dps_val: false
+          icon: mdi:air-humidifier-off
+          icon_priority: 2
+    - id: 13
+      name: humidity
+      type: integer
+      range:
+        min: 30
+        max: 75
+        step: 5
+    - id: 14
+      name: current_humidity
+      type: integer
+    - id: 23
+      type: string
+      name: mode
+      mapping:
+        - dps_val: auto
+          value: auto
+          icon: mdi:refresh-auto
+          icon_priority: 3
+        - dps_val: level1
+          value: low
+          icon: mdi:speedometer-slow
+          icon_priority: 3
+        - dps_val: level2
+          value: middle
+          icon: mdi:speedometer-medium
+          icon_priority: 3
+        - dps_val: level3
+          value: high
+          icon: mdi:speedometer
+          icon_priority: 3
+        - dps_val: sleep
+          value: sleep
+          icon: mdi:sleep
+          icon_priority: 3
+secondary_entities:
+  - entity: switch
+    name: Sound
+    dps:
+      - id: 8
+        name: switch
+        type: boolean
+        mapping:
+          - dps_val: true
+            icon: mdi:volume-high
+          - dps_val: false
+            icon: mdi:volume-off
+  - entity: sensor
+    name: Current Temperature
+    class: temperature
+    dps:
+      - id: 10
+        type: integer
+        name: sensor
+        unit: C
+  - entity: select
+    name: Timer
+    icon: mdi:timer
+    dps:
+      - id: 19
+        name: option  
+        type: string
+        mapping:
+          - dps_val: cancel
+            value: 'Off'
+          - dps_val: 1h
+            value: 1 hour
+          - dps_val: 2h
+            value: 2 hours
+          - dps_val: 3h
+            value: 3 hours
+          - dps_val: 4h
+            value: 4 hours
+          - dps_val: 5h
+            value: 5 hours
+          - dps_val: 6h
+            value: 6 hours
+          - dps_val: 7h
+            value: 7 hours
+          - dps_val: 8h
+            value: 8 hours
+          - dps_val: 9h
+            value: 9 hours
+          - dps_val: 10h
+            value: 10 hours
+          - dps_val: 11h
+            value: 11 hours
+          - dps_val: 12h
+            value: 12 hours
+      - id: 20
```

Comment:
Time remaining can be made a `class: duration` sensor, it is generally interesting enough to have as a sensor rather than an extra attribute where it is more difficult to get at.